dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/matcher/ProxyClassIgnores.java

```diff
@@ -5,16 +5,25 @@ public class ProxyClassIgnores {
   private ProxyClassIgnores() {}
 
   public static boolean isIgnored(String name) {
-    if (name.indexOf('$') > -1) {
-      if (name.contains("$JaxbAccessor")
-          || name.contains("CGLIB$$")
-          || name.contains("$__sisu")
-          || name.contains("$$EnhancerByGuice$$")
-          || name.contains("$$EnhancerByProxool$$")
-          || name.contains("$$$view")
-          || name.contains("$$$endpoint") // jboss mdb proxies
-          || name.contains("$$_Weld")
-          || name.contains("_$$_jvst")) {
+    int last = -1;
+    int idx;
+    while (true) {
+      idx = name.indexOf('$', last + 1);
```

Comment:
```suggestion
    for (int last = -1, idx; (idx = name.indexOf('$', last + 1)) >= 0; last = idx) {
```
This should avoid using infinite loop.
You can also remove the `last = id` in the `continue` and default branches.

Reply:
Thanks it's safer and works. I had to remove also the last assignement so I added a commit for it.